common/src/main/java/org/apache/sedona/common/raster/RasterAccessors.java

```diff
@@ -75,12 +80,62 @@ public static double getScaleY(GridCoverage2D raster) {
         return RasterUtils.getGDALAffineTransform(raster).getScaleY();
     }
 
+    public static double getSkewX(GridCoverage2D raster) {
+        return RasterUtils.getGDALAffineTransform(raster).getShearX();
+    }
+
+    public static double getSkewY(GridCoverage2D raster) {
+        return RasterUtils.getGDALAffineTransform(raster).getShearY();
+    }
+
     public static double getWorldCoordX(GridCoverage2D raster, int colX, int rowY) throws TransformException {
-        return RasterUtils.getCornerCoordinates(raster, colX, rowY).getX();
+        return RasterUtils.getWorldCornerCoordinates(raster, colX, rowY).getX();
     }
 
     public static double getWorldCoordY(GridCoverage2D raster, int colX, int rowY) throws TransformException {
-        return RasterUtils.getCornerCoordinates(raster, colX, rowY).getY();
+        return RasterUtils.getWorldCornerCoordinates(raster, colX, rowY).getY();
+    }
+
+    public static Geometry getGridCoord(GridCoverage2D raster, double longitude, double latitude) throws TransformException {
```

Comment:
Given that the CRS transformation follows the lat/lon order, the parameters here are wrong. To avoid unnecessary complication, please call them X/Y, instead of longitude and latitude.

Reply:
Got it, changed lon and lat to x and y respectively